### PR TITLE
Support pf upgrade

### PR DESCRIPTION
**Pull request by YingChen1996**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

win pypi:
image

win msi:
image

linux pypi:
image

linux script:
image

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_cli/_pf/_upgrade.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        return
+
+    yes = args.yes
+    exit_code = 0
+    installer = os.getenv(_ENV_PF_INSTALLER) or ''
+    installer = installer.upper()
+    print(f"installer: {installer}")
+    latest_version_msg = 'It will be updated to {}.'.format(latest_version) if yes \
+        else 'Latest version available is {}.'.format(latest_version)
+    logger.warning("Your current Azure CLI version is %s. %s", local_version, latest_version_msg)
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_cli/_pf/_upgrade.py:62]: Azure CLI -> pf CLI

File: src/promptflow/promptflow/_cli/_pf/_upgrade.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+            logger.debug('No tty available.')
+            raise UserErrorException("No tty available. Please run command with --yes.")
+        confirmation = prompt_y_n("Do you want to continue?", default='y')
+
+        if not confirmation:
+            logger.debug("Upgrade stopped by user")
+            return
+
+    if installer == 'MSI':
+        _upgrade_on_windows()
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_cli/_pf/_upgrade.py:75]: can we pass in the yes arg to msi install comamnd?

File: src/promptflow/promptflow/_cli/_pf/_upgrade.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    importlib.reload(json)
+
+    version_result = subprocess.check_output(['pf', 'version'], shell=platform.system() == 'Windows')
+    version_json = json.loads(version_result)
+    new_version = version_json['promptflow']
+
+    if new_version == local_version:
+        logger.warning(f"new_version: {new_version}, local_version: {local_version}")
+        err_msg = "CLI upgrade failed or aborted."
+        logger.warning(err_msg)
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_cli/_pf/_upgrade.py:107]: CLI upgrade to version {latest_version } failed or aborted.

File: src/promptflow/promptflow/_cli/_pf/_upgrade.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    import json
+    importlib.reload(subprocess)
+    importlib.reload(json)
+
+    version_result = subprocess.check_output(['pf', 'version'], shell=platform.system() == 'Windows')
+    version_json = json.loads(version_result)
+    new_version = version_json['promptflow']
+
+    if new_version == local_version:
+        logger.warning(f"new_version: {new_version}, local_version: {local_version}")
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_cli/_pf/_upgrade.py:105]: why do we need to print this line?

File: src/promptflow/promptflow/_cli/_pf/_upgrade.py (hunk 1 above)
Comment (wangchao1230) [reply src/promptflow/promptflow/_cli/_pf/_upgrade.py:62]: Upgrading pf CLI version: {local_version} -> {latest_version}

File: src/promptflow/promptflow/_cli/_pf/_upgrade.py (hunk 2 above)
Comment (YingChen1996) [reply src/promptflow/promptflow/_cli/_pf/_upgrade.py:75]: why we need pass yes to msi?

**Review by wangchao1230 — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Review by crazygao — APPROVED** [review]

**Pull request merged**